backend/TreeOfAKind.Infrastructure/FileStorage/AzureBlobStorageService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using MimeTypes;
+using Serilog;
+using TreeOfAKind.Application.Services;
+using TreeOfAKind.Domain.Trees;
+
+namespace TreeOfAKind.Infrastructure.FileStorage
+{
+    public class AzureBlobStorageService : IFileSaver
+    {
+        private readonly ILogger _logger;
+        private readonly IDictionary<string, string> _metadata;
+        private readonly BlobServiceClient _blobServiceClient;
+
+        private const string Wildcard = "*";
+
+
+        private static readonly BlobRequestConditions BlobRequestConditions
+            = new BlobRequestConditions {IfNoneMatch = new ETag(Wildcard)};
+
+        public AzureBlobStorageService(AzureBlobStorageSettings settings, ILogger logger)
+        {
+            _logger = logger;
+            var connectionString = settings.ConnectionString;
+            _metadata = settings.Metadata;
+
+            _blobServiceClient = new BlobServiceClient(connectionString);
+        }
+
+        public async Task<Uri> UploadFile(string containerName, string contentType, Stream stream,
```

Comment:
Also you don't have to provide that generic interface, it'll bite when you'll implement adding documents, but it's fine for now.